core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -120,22 +120,25 @@ oppia.factory('NumberWithUnitsObjectFactory', [
             units = rawInput.substr(ind).trim();
           }
         } else {
-          if (!rawInput.startsWith('$ ') && !rawInput.startsWith('Rs ') &&
-            !rawInput.startsWith('₹ ')) {
+          if (!rawInput.startsWith('$') && !rawInput.startsWith('Rs') &&
+            !rawInput.startsWith('₹')) {
             throw new Error(NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY);
           }
-          var ind = rawInput.indexOf(' ');
+          var ind = rawInput.indexOf(rawInput.match(/[0-9]/));
           if (ind === -1) {
             throw new Error(NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY);
           }
-          units = rawInput.substr(0, ind) + ' ';
+          units = rawInput.substr(0, ind).trim() + ' ';
+          if (!(units === '$ ') && !(units === 'Rs ') && !(units === '₹ ')) {
+            throw new Error(NUMBER_WITH_UNITS_PARSING_ERRORS.INVALID_CURRENCY);
+          }
           var ind2 = rawInput.indexOf(
             rawInput.substr(ind + 1).match(/[a-z(]/i));
```

Comment:
Why not start from `ind` itself? You have removed `+1` from other places.

Reply:
done